src/_pytest/pytester.py

```diff
@@ -47,13 +51,17 @@
 from _pytest.python import Module
 from _pytest.reports import CollectReport
 from _pytest.reports import TestReport
-from _pytest.tmpdir import TempdirFactory
+from _pytest.tmpdir import TempPathFactory
 
 if TYPE_CHECKING:
     from typing_extensions import Literal
 
     import pexpect
 
+    PathLike = os.PathLike[str]
```

Comment:
This is a bit confusing, I think it will be less confusing to use `"os.PathLike[str]"` quoted in the type annotations.